pkg/client/client.go

```diff
@@ -52,6 +52,8 @@ type ImageFetcher interface {
 	// PullIfNotPresent and daemon = false, gives us the same behavior as PullAlways.
 	// There is a single invalid configuration, PullNever and daemon = false, this will always fail.
 	Fetch(ctx context.Context, name string, options image.FetchOptions) (imgutil.Image, error)
+
+	CheckReadAccessValidator(options image.FetchOptions) image.CheckReadAccess
```

Comment:
I think you are making this a method that returns a function because you want to provide an "access checker" to `getBestRunMirror`. But why not just pass the whole image fetcher to that function? You don't need to expect an image fetcher, just an interface that satisfies `CheckReadAccess`.

Reply:
Well, I wanted to preserve the signature for CheckReadAccess function `func(repo string) bool`, but yes I think we can pass through the `Fetcher` instance